modules/swagger-codegen/src/main/resources/Java/libraries/retrofit2/ApiClient.mustache

```diff
@@ -154,40 +156,8 @@ public class ApiClient {
       .create(serviceClass);
   }
 
-  public ApiClient setDateFormat(DateFormat dateFormat) {
-    this.json.setDateFormat(dateFormat);
-    return this;
-  }
-
-  public ApiClient setSqlDateFormat(DateFormat dateFormat) {
-    this.json.setSqlDateFormat(dateFormat);
-    return this;
-  }
-
-  {{#joda}}
-  public ApiClient setDateTimeFormat(DateTimeFormatter dateFormat) {
-    this.json.setDateTimeFormat(dateFormat);
-    return this;
-  }
-
-  public ApiClient setLocalDateFormat(DateTimeFormatter dateFormat) {
-    this.json.setLocalDateFormat(dateFormat);
-    return this;
-  }
-
-  {{/joda}}
-  {{#jsr310}}
-  public ApiClient setOffsetDateTimeFormat(DateTimeFormatter dateFormat) {
-    this.json.setOffsetDateTimeFormat(dateFormat);
-    return this;
-  }
-
-  public ApiClient setLocalDateFormat(DateTimeFormatter dateFormat) {
-    this.json.setLocalDateFormat(dateFormat);
-    return this;
-  }
-
-  {{/jsr310}}
+{{#gson}}{{>gson/mappingConverterFactory}}{{/gson}}
```

Comment:
Can we put this into another class/file instead ?

Reply:
Eg : the JSON class

Reply:
I have fixed the java6 issue, but I don't believe that this change will work.  The JSON class for the gson library was necessary for all the gson configuration but Jackson doesn't need any configuration in this circumstance, just to replace the gson logic in this class which is tied to the ApiClient

Reply:
There is also mapper configuration for Jackson that should be moved to a JSON class.